.github/actions/get-bundles/action.yml

```diff
@@ -0,0 +1,109 @@
+# questions.
+#
+
+name: 'Get bundles'
+description: 'Download resulting JDK bundles'
+inputs:
+  platform:
+    description: 'Platform name'
+    required: true
+  debug-suffix:
+    description: 'File name suffix denoting debug level, possibly empty'
+    required: false
+outputs:
+  jdk-path:
+    description: 'Path to the installed JDK bundle'
+    value: ${{ steps.path-name.outputs.jdk }}
+  symbols-path:
+    description: 'Path to the installed symbols bundle'
+    value: ${{ steps.path-name.outputs.symbols }}
+  tests-path:
+    description: 'Path to the installed tests bundle'
+    value: ${{ steps.path-name.outputs.tests }}
+
+runs:
+  using: composite
+  steps:
+    - name: 'Download bundles artifact'
+      id: download-bundles
+      uses: actions/download-artifact@v3
+      with:
+        name: bundles-${{ inputs.platform }}${{ inputs.debug-suffix }}
+        path: bundles
+      continue-on-error: true
+
+    - name: 'Download bundles artifact (retry)'
+      uses: actions/download-artifact@v3
+      with:
+        name: bundles-${{ inputs.platform }}${{ inputs.debug-suffix }}
+        path: bundles
+      if: steps.download-bundles.outcome == 'failure'
+
+    - name: 'Unpack bundles'
+      run: |
+        if [[ -e bundles/jdk-${{ inputs.platform }}${{ inputs.debug-suffix }}.zip ]]; then
+          echo 'Unpacking jdk bundle...'
+          mkdir -p bundles/jdk
+          unzip -q bundles/jdk-${{ inputs.platform }}${{ inputs.debug-suffix }}.zip -d bundles/jdk
+        fi
+
+        if [[ -e bundles/jdk-${{ inputs.platform }}${{ inputs.debug-suffix }}.tar.gz ]]; then
+          echo 'Unpacking jdk bundle...'
+          mkdir -p bundles/jdk
+          tar -xf bundles/jdk-${{ inputs.platform }}${{ inputs.debug-suffix }}.tar.gz -C bundles/jdk
+        fi
+
+        if [[ -e bundles/symbols-${{ inputs.platform }}${{ inputs.debug-suffix }}.tar.gz ]]; then
+          echo 'Unpacking symbols bundle...'
+          mkdir -p bundles/symbols
+          tar -xf bundles/symbols-${{ inputs.platform }}${{ inputs.debug-suffix }}.tar.gz -C bundles/symbols
+        fi
+
+        if [[ -e bundles/tests-${{ inputs.platform }}${{ inputs.debug-suffix }}.tar.gz ]]; then
+          echo 'Unpacking tests bundle...'
+          mkdir -p bundles/tests
+          tar -xf bundles/tests-${{ inputs.platform }}${{ inputs.debug-suffix }}.tar.gz -C bundles/tests
+        fi
+      shell: bash
+
+    - name: 'Export paths to where bundles are installed'
+      id: path-name
+      run: |
+        # Export the paths
+
+        jdk_dir="$GITHUB_WORKSPACE/$(dirname $(find bundles/jdk -name bin -type d))"
+        symbols_dir="$GITHUB_WORKSPACE/$(dirname $(find bundles/symbols -name bin -type d))"
+        tests_dir="$GITHUB_WORKSPACE/bundles/tests"
+
+        if [[ '${{ runner.os }}' == 'Windows' ]]; then
+          jdk_dir="$(cygpath $jdk_dir)"
+          symbols_dir="$(cygpath $symbols_dir)"
+          tests_dir="$(cygpath $tests_dir)"
```

Comment:
Here, and maybe later: how safe it is to use Cygwin utilities like `cygpath` in MSYS2 context?

Reply:
Perfectly safe and officially supported. MSYS2 is related to MSYS(1) by name only, it is in reality a fork/clone of Cygwin. The principal difference is that MSYS2 is more pragmatic when it comes to choosing between perfect POSIX compliance, or have tooling that is helpful in creating a POSIX-like build environment on Windows.